### ISSUE TITLE
pdnsutil edit-zone not working on EL7 by default

### ISSUE THREAD

**Issue by hnsk** [PowerDNS/pdns#4173]
EL7 environment doesn't have `which editor` or $EDITOR set by default.

This causes the following to happen:
# pdnsutil edit-zone vice.fi
Jul 14 10:47:09 Reading random entropy from '/dev/urandom'
Jul 14 10:47:09 gmysql Connection successful. Connected to database 'poweradmin' on '127.0.0.1'.
Jul 14 10:47:09 gmysql Connection successful. Connected to database 'poweradmin' on '127.0.0.1'.
sh: editor: command not found
Error: Editing file with: 'editor /tmp/pdnsutil-Ly1gwm', perhaps set EDITOR variable: Success

Also the ":Success" in the end is slightly misleading as it succeeded in erroring.

Maybe it should default to vi(m) by default as that is what most applications do (visudo for example) and if someone prefers something else then they most likely have $EDITOR set.

**Comment by jpmens** [comment]
FWIW (and I know this because it upsets me when I forget to configure `$EDITOR` on new boxes), Debian defaults to `nano` (e.g. in _git_, _visudo_, etc.).

**Comment by cmouse** [comment]
git uses
```
/usr/bin/git-core/editor
/usr/local/bin/editor
/usr/bin/editor
```
until it finds something. Possibly uses something else as well.

**Comment by hnsk** [comment]
https://github.com/PowerDNS/pdns/pull/4176 fixes this by defaulting to vi if $EDITOR is not set or editor found in path.

**Comment by slowaways** [comment]
I have this error now:

sh: editor: command not found
Error: Editing file with: 'editor /tmp/pdnsutil-5VVukw', perhaps set EDITOR variable: Resource temporarily unavailable

**Comment by jpmens** [comment]
Have you read the actual error message? If you have, which steps did you undertake to solve the problem?

**Comment by slowaways** [comment]
> 
> 
> Have you read the actual error message? If you have, which steps did you undertake to solve the problem?

I did not understand, sorry.

**Comment by cmouse** [comment]
`sh: editor: command not found`

**Comment by slowaways** [comment]
> 
> 
> `sh: editor: command not found`

Yes. I already tried to define an alias to "editor" or define the editor variable for the nano editor path, but it doesn't work.

**Comment by jpmens** [comment]
The variable is called `EDITOR`.

```console
$ EDITOR=/path/to/your/favorite/editing/utility
$ export EDITOR
```

If that doesn't work: show us

```console
$ which nano
```

**Comment by MTRNord** [comment]
Hi I am hitting the same issue on nixos. even after setting $EDITOR I get the "editor: command not found" error.



```

[nix-shell:~]$ which nano
/run/current-system/sw/bin/nano

[nix-shell:~]$ export EDITOR=/run/current-system/sw/bin/nano

[nix-shell:~]$ EDITOR=/run/current-system/sw/bin/nano

[nix-shell:~]$ export EDITOR

[nix-shell:~]$ which nano
/run/current-system/sw/bin/nano

[nix-shell:~]$ ls -la /run/current-system/sw/bin/nano
lrwxrwxrwx 6 root root 61 Jan  1  1970 /run/current-system/sw/bin/nano -> /nix/store/xwcfd3mfin810fadq4nqaaihl1cyrqwr-nano-7.2/bin/nano

[nix-shell:~]$ sudo pdnsutil edit-zone midnightthoughts.space --config-dir=/run/pdns
sh: line 1: editor: command not found
Error: Editing file with: 'editor /tmp/pdnsutil-fcyTgq', perhaps set EDITOR variable: No such file or directory
```

**Comment by buergi** [comment]
I pretty much assume this issue can be closed, either EDITOR was not set or chroot is used and the editor is not inside the chroot.
